projects/packages/forms/src/blocks/form-step-navigation/edit.js

```diff
@@ -0,0 +1,206 @@
+	},
+];
+export const NEXT_BUTTON_TEMPLATE = [
+	'jetpack/button',
+	{
+		element: 'button',
+		text: __( 'Next', 'jetpack-forms' ),
+		uniqueId: 'next-step',
+		customVariant: 'next',
+		metaName: __( 'Next button', 'jetpack-forms' ),
+	},
+];
+
+const SUBMIT_BUTTON_TEMPLATE = [
+	'jetpack/button',
+	{
+		element: 'button',
+		text: __( 'Submit', 'jetpack-forms' ),
+		uniqueId: 'submit-step',
+		customVariant: 'submit',
+		metaName: __( 'Submit button', 'jetpack-forms' ),
+	},
+];
+
+export const NAVIGATION_TEMPLATE = [
+	PREVIOUS_BUTTON_TEMPLATE,
+	NEXT_BUTTON_TEMPLATE,
+	SUBMIT_BUTTON_TEMPLATE,
+];
+
+export default function Edit( { clientId } ) {
+	const blockProps = useBlockProps();
+
+	const { replaceInnerBlocks, __unstableMarkNextChangeAsNotPersistent } =
+		useDispatch( blockEditorStore );
+
+	const { ancestorStepClientId } = useSelect(
+		select => {
+			const { getBlockParentsByBlockName } = select( blockEditorStore );
+			const stepParentArray = getBlockParentsByBlockName( clientId, [ 'jetpack/form-step' ] );
+			return {
+				ancestorStepClientId: stepParentArray.length > 0 ? stepParentArray[ 0 ] : null,
+			};
+		},
+		[ clientId ]
+	);
+
+	const formClientId = useParentFormClientId( clientId );
+	const steps = useFormSteps( formClientId );
+
+	// Get the single step mode state and active step
+	const { isSingleStep, activeStepId } = useSelect(
+		select => {
+			if ( ! formClientId ) return { isSingleStep: false, activeStepId: null };
+			const { isSingleStepMode, getActiveStepId } = select( singleStepStore );
+			return {
+				isSingleStep: isSingleStepMode( formClientId ),
+				activeStepId: getActiveStepId( formClientId ),
+			};
+		},
+		[ formClientId ]
+	);
+
+	// Check if we're inside a step or standalone
+	const isOutsideSteps = ! ancestorStepClientId;
+
+	// Track the current step index (used later when replacing inner blocks)
+	let currentIndex = 0;
+
+	if ( isOutsideSteps && isSingleStep && activeStepId ) {
+		// When outside steps but in single step mode, show buttons based on the active step
+		const activeStepIndex = steps.findIndex( block => block.clientId === activeStepId );
+		if ( activeStepIndex !== -1 ) {
+			currentIndex = activeStepIndex;
+		}
+	} else if ( ! isOutsideSteps ) {
+		// Inside a step - determine position
+		const stepIndex = steps.findIndex( block => block.clientId === ancestorStepClientId );
+		currentIndex = stepIndex;
+	}
+
+	const { navigationBlocks } = useSelect(
+		select => {
+			const { getBlocks } = select( blockEditorStore );
+			const currentNavigationBlocks = getBlocks( clientId );
+			return {
+				navigationBlocks: currentNavigationBlocks,
+			};
+		},
+		[ clientId ]
+	);
+
+	const innerBlocksProps = useInnerBlocksProps( blockProps, {
+		template: NAVIGATION_TEMPLATE,
+		templateLock: 'all',
+		allowedBlocks: [ 'jetpack/button' ],
+		renderAppender: false,
+	} );
+
+	useEffect( () => {
+		// This happends when the step block is duplilcated
```

Comment:
Couple of typos. I think with more complicated effects it might be worth putting a brief description of what the effect does as well, as that will help other devs get an idea more quickly.

```suggestion
		// This happens when the step block is duplicated
```